tests/ScreenshotPage.test.js

```diff
@@ -0,0 +1,309 @@
+import { render, screen, fireEvent } from '@testing-library/react'
+import React from 'react'
+import ScreenshotPage from '../src/components/game/ScreenshotPage'
+import sampleScreenshot from './image'
+import Context from '../src/context/context'
+
+/**
+ * A custom render to setup providers. Extends regular
+ * render options with `providerProps` to allow injecting
+ * different scenarios to test with.
+ */
+const customRender = (ui, { providerProps, ...renderOptions }) => {
```

Comment:
+1 for using React Testing Library. I'm a huge fan of black-box testing for UIs which RTL promotes (I'm sure @Dondi probably uses this too now)